nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/CompareFuzzyHash.java

```diff
@@ -0,0 +1,289 @@
+                fuzzyHashMatcher = new SSDeepHashMatcher(getLogger());
+                break;
+        }
+
+        if (fuzzyHashMatcher.isValidHash(inputHash) == false) {
+            // and if that is the case we log
+            logger.error("Invalid hash provided. Sending to failure");
+            //  and send to failure
+            session.transfer(flowFile, REL_FAILURE);
+            session.commit();
+            return;
+        }
+
+        File file = new File(context.getProperty(HASH_LIST_FILE).getValue());
+
+        double similarity = 0;
+        double matchThreshold = context.getProperty(MATCH_THRESHOLD).asDouble();
+
+        try {
+            Map<String, Double> matched = new ConcurrentHashMap<String, Double>();
+            FileInputStream fileInputStream = new FileInputStream(file);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(fileInputStream));
+
+            // If SSdeep skip the first line (as the usual format used by other tools add a header line
+            // to a file list
+            if (algorithm == ssdeep) {
+                reader.readLine();
+            }
+
+            String line = null;
+            String[] hashToCompare = null;
+
+            iterateFile: while ((line = reader.readLine()) != null) {
+                switch (context.getProperty(HASH_ALGORITHM).getValue()) {
+                    case tlsh:
+                        hashToCompare = line.split("\t", 2);
+                        break;
+                    case ssdeep:
+                        hashToCompare = line.split(",", 2);
+                        break;
+                }
+
+                if (hashToCompare != null) {
+                    similarity = fuzzyHashMatcher.getSimilarity(inputHash, hashToCompare[0]);
+
+                    if (fuzzyHashMatcher.matchExceedsThreshold(similarity, matchThreshold)) {
+                        //
+                        matched.put(hashToCompare[1], similarity);
+                    }
+                }
+
+                // Check if single match is desired and if a match has been made
+                if (context.getProperty(MATCHING_MODE).getValue() == singleMatch.getValue() && (matched.size() > 0)) {
+                    // and save time by breaking the outer loop
+                    break iterateFile;
+                }
+            }
+            // no matter if the break was called or not, Continue processing
+            // First by creating a new map to hold attributes
+            Map<String, String> attributes = new ConcurrentHashMap<String, String>();
+
+            // Then by iterating over the hashmap of matches
+            if (matched.size() > 0) {
+                int x = 0;
+                for (Map.Entry<String, Double> entry : matched.entrySet()) {
+                    // defining attributes accordingly
+                    attributes.put(
+                            context.getProperty(ATTRIBUTE_NAME).getValue() + "." + x + ".match",
```

Comment:
Use **Extract Variable** refactor on `context.getProperty(ATTRIBUTE_NAME).getValue() + "." + x` to avoid multiple calls to two external objects and repeated String concatenation.